### DISCUSSION TITLE
Force the Y value of all digits to be the same

### DISCUSSION THREAD

**Discussion by caco3** [jomjol/AI-on-the-edge-device#1130] [Ideas] [answered]
@jomjol @haverland 

We often see that people do not align all digits of a meter to the same Y value, eg. in https://github.com/jomjol/AI-on-the-edge-device/discussions/1128

Would it make sense to lock all Y values of a digital meter to the same Y value? That would make it easier during setup and also prevent misalignment.

Also, as a 2nd thought, in most meters I have seen, the X distance between digital digits is the always the same. We also could enforce this.

**Comment by ghost** [comment]
Cool Idea. I could imagine a button make all other ROIs the same y and make all other ROIs the same dimensions. 

I'm not with you on the x. The distance of the camera depends on the printing. I use here watermeter prints with different spacers and the electric meter with mirror. So the X distance is not every time the same.

**Reply by caco3** [reply]
We could implement it in javascript, then its not intrusive.

What about 2 checkboxes:
 - [x] All ROI's on same Y level
 - [x] [Equidistanct](https://en.wikipedia.org/wiki/Equidistant_set) ROI's on X axis

By default I would keep them set and people can untick for traditional setup.

**Comment by cbho** [comment, marked as answer]
Hi,
I totally support the idea about making it possible to align ROIs.
However, I think this could cause confusion given the instructions on how to define [ROIs in the wiki](https://github.com/jomjol/AI-on-the-edge-device/wiki/ROI-Configuration).

When having rolling numbers, they often are _not_ aligned, when you are setting up the device.
Wouldn't aligning the ROIs on a device with rolling numbers with a great propability cause the areas to be defined on only parts of one or more of the numbers? Wouldn't that cause trouble? (Especially in the beginning - after some time, when numbers have rotated, I guess things will sort out :-) )

**Reply by caco3** [reply]
There are now optional sync features contained.
For whoever does not want or can use them, they can be disabled.
